static/app/views/performance/queues/messageConsumerSamplesPanel.tsx

```diff
@@ -0,0 +1,273 @@
+                  >
+                    {query.transaction}
+                  </Link>
+                </Title>
+              </TitleContainer>
+            </HeaderContainer>
+          </ModuleLayout.Full>
+
+          <ModuleLayout.Full>
+            <MetricsRibbon>
+              <MetricReadout
+                align="left"
+                title={t('Processed')}
+                value={transactionMetrics?.[0]?.['count()']}
+                unit={'count'}
+                isLoading={aretransactionMetricsFetching}
+              />
+              <MetricReadout
+                align="left"
+                title={t('Error Rate')}
+                value={undefined}
+                unit={'percentage'}
+                isLoading={aretransactionMetricsFetching}
+              />
+              <MetricReadout
+                title={t('Avg Time In Queue')}
+                value={undefined}
+                unit={DurationUnit.MILLISECOND}
+                isLoading={false}
+              />
+              <MetricReadout
+                title={t('Avg Processing Latency')}
+                value={
+                  transactionMetrics[0]?.[
+                    'avg_if(span.self_time,span.op,queue.task.celery)'
+                  ]
+                }
+                unit={DurationUnit.MILLISECOND}
+                isLoading={false}
+              />
+            </MetricsRibbon>
+          </ModuleLayout.Full>
+          <ModuleLayout.Full>
+            <DurationChart
+              series={[
+                {
+                  ...durationData[`avg(span.self_time)`],
+                  markLine: AverageValueMarkLine(),
+                },
+              ]}
+              scatterPlot={sampledSpanDataSeries}
+              onHighlight={highlights => {
+                const firstHighlight = highlights[0];
+
+                if (!firstHighlight) {
+                  setHighlightedSpanId(undefined);
+                  return;
+                }
+
+                const sample = findSampleFromDataPoint(firstHighlight.dataPoint);
+                setHighlightedSpanId(sample?.span_id);
+              }}
+              isLoading={isDurationDataFetching}
+              error={durationError}
+            />
+          </ModuleLayout.Full>
+
+          <ModuleLayout.Full>
+            <MessageSpanSamplesTable
+              data={durationSamplesData}
+              isLoading={isDurationDataFetching || isDurationSamplesDataFetching}
+              highlightedSpanId={highlightedSpanId}
+              onSampleMouseOver={sample => setHighlightedSpanId(sample.span_id)}
+              onSampleMouseOut={() => setHighlightedSpanId(undefined)}
+              error={durationSamplesDataError}
+              // Samples endpoint doesn't provide meta data, so we need to provide it here
+              meta={{
+                fields: {
+                  'span.self_time': 'duration',
+                },
+                units: {},
+              }}
```

Comment:
Should we be typing the units on this as well? I guess the default unit for duration is `ms` so I'm okay if it's left out as well.